core/src/main/java/com/yahoo/imapnio/async/data/MessageNumberSet.java

```diff
@@ -257,4 +257,40 @@ public static String buildString(@Nullable final MessageNumberSet[] msgsets) {
         }
         return s.toString();
     }
+
+    /**
+     * Converts an IMAP RFC3501 sequence-set syntax into an array of MessageNumberSet.
+     *
+     * @param msgNumbers the message numbers string
+     * @return the array of MessageNumberSet
+     * @throws ImapAsyncClientException will not throw
+     */
+    public static MessageNumberSet[] buildMessageNumberSets(@Nullable final String msgNumbers) throws ImapAsyncClientException {
+        if (msgNumbers == null || msgNumbers.isEmpty()) {
+            return null;
+        }
+
+        final String[] msgSetStrs = msgNumbers.split(",");
+        final MessageNumberSet[] msgSets = new MessageNumberSet[msgSetStrs.length];
+
+        int i = 0; // msgset index
+        for (final String element : msgSetStrs) {
+            if (element.contains(":")) {
+                final String[] elements = element.split(":");
+                if (elements[1].equals("*")) { // Ex: 1:*
+                    msgSets[i] = new MessageNumberSet(Long.valueOf(elements[0]), LastMessage.LAST_MESSAGE);
+                } else { // Ex: 1:2
+                    msgSets[i] = new MessageNumberSet(Long.valueOf(elements[0]), Long.valueOf(elements[1]));
+                }
+            } else {
+                if (element.equals("*")) { // Ex: *
+                    msgSets[i] = new MessageNumberSet(LastMessage.LAST_MESSAGE);
+                } else { // Ex: 1
+                    msgSets[i] = new MessageNumberSet(Long.valueOf(element), Long.valueOf(element));
```

Comment:
Assign the value of Long.valueOf(element) to a tmp variable, so you don't have to parse string twice.